colossalai/tensor/moe_tensor/moe_info.py

```diff
@@ -15,12 +15,8 @@ def __init__(self, ep_inside: bool, ep_size: int, dp_size: int, pp_size: int = 1
             ep_inside (bool, optional): Use ep inside dp if True, dp inside ep if Fasle. Defaults to True.
         """
         self.pp_size, self.dp_size, self.ep_size = pp_size, dp_size, ep_size
-        if ep_inside:
-            self.pp_axis, self.dp_axis, self.ep_axis = 0, 1, 2
-            self.pg = ProcessGroupMesh(self.pp_size, self.dp_size, self.ep_size)
-        else:
-            self.pp_axis, self.ep_axis, self.dp_axis = 0, 1, 2
-            self.pg = ProcessGroupMesh(self.pp_size, self.ep_size, self.dp_size)
+        self.pp_axis, self.dp_axis, self.ep_axis = (0, 1, 2) if ep_inside else (0, 2, 1)
+        self.pg = ProcessGroupMesh(self.pp_size, self.dp_size, self.ep_size)
```

Comment:
Process group should support dp inside or ep size inside. You can add an assert if hierachical alltoall does not support one of them

Reply:
Hierarchical all-to-all communication can support both of the mentioned scenarios. As long as the experts are sharded by their rank in ascending order. For example, if `ep_group = [0, 1, 8, 9]`, then rank 0 will contain expert shard 0, rank 1 will contain expert shard 1, and so on.

Reply:
You fix the processgroupmesh order in `self.pg = ProcessGroupMesh(self.pp_size, self.dp_size, self.ep_size)`. It should change with dp and ep axis

Reply:
I changed it unintentionally. d93b7709 revert this.